### PR TITLE
opt setCurrentStream function

### PR DESCRIPTION
**Pull request by hellozmz**
## Motivation and Context

Improve performance by reducing the number of calls to the setCurStream() function.

ref: #1224

## Description

降低函数setCurStream的调用量。在resnet50上进行测试，性能有提升。通过对优化之前的代码和优化之后的代码进行测试，展示优化后的效果，整体进行了两轮测试。
纵坐标Time：模型运行1个step耗时。
横坐标Step。
橙色是优化后的代码，蓝色是优化前的代码。整体趋势上优化后耗时有降低，性能有提升。

image
image

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/torch/helper.hpp [hunk 1]
```diff
@@ -53,11 +53,23 @@ namespace impl {
 
 namespace aten {
 
+
+static const size_t MAX_GPU_NUMS = 32;
```
Comment (lljbash) [inline impl/torch/helper.hpp:57]:
```suggestion
constexpr size_t MAX_GPU_NUMS = 16;
```
而且也可以放到函数里。

貌似最大值是 16？dipu 里面是。

File: impl/torch/helper.hpp [hunk 2]
```diff
@@ -53,11 +53,23 @@ namespace impl {
...
+
     diopiStreamHandle_t stream_handle;
     diopiGetStream(ctx, &stream_handle);
-    c10::cuda::CUDAStream cur_stream = c10::cuda::getStreamFromExternal(static_cast<cudaStream_t>(stream_handle), c10::cuda::current_device());
-    c10::cuda::setCurrentCUDAStream(cur_stream);
+
+    int device_id = c10::cuda::current_device();
+    TORCH_CHECK(device_id >= 0 && device_id < MAX_GPU_NUMS, "device_id is out of range");
+
+    if (!current_streams[device_id] || current_streams[device_id] != stream_handle) {
```
Comment (lljbash) [inline impl/torch/helper.hpp:68]: 第一个检查可以去掉吧，你零初始化了，下一个检查也过不去
要是空的传进来说明哪里不对了，就让他炸

**Review by lljbash — APPROVED** [review]

File: impl/torch/helper.hpp (hunk 2 above)
Comment (hellozmz) [reply impl/torch/helper.hpp:68]: > 第一个检查可以去掉吧，你零初始化了，下一个检查也过不去 要是空的传进来说明哪里不对了，就让他炸

好的

File: impl/torch/helper.hpp (hunk 1 above)
Comment (hellozmz) [reply impl/torch/helper.hpp:57]: > 而且也可以放到函数里。
> 
> 貌似最大值是 16？dipu 里面是。

这个我调整下。

**Review by yangbofun — APPROVED** [review]

**Pull request merged**